src/Application.vala

```diff
@@ -103,12 +103,57 @@ public class Music.Application : Gtk.Application {
         settings.bind ("window-maximized", main_window, "maximized", SettingsBindFlags.SET);
     }
 
+    private File[] list_directory (string directory) {
+        try {
+            var dir = Dir.open (directory, 0);
+
+            string? name = null;
+            File[] elements = {};
+
+            while ((name = dir.read_name ()) != null) {
+                string file_path = Path.build_filename (directory, name);
+                File file = File.new_for_path (file_path);
+
+                elements += file;
+            }
+
+            return elements;
+        } catch (FileError e) {
```

Comment:
Can you put only `var dir = Dir.open (directory, 0);` in try-catch please?

Reply:
Won't everything else execute then anyway if something goes wrong?

Reply:
Done